Application/mobile/app_code/lib/pages/container/details_state.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+import 'package:provider/provider.dart';
+import 'package:risu/components/alert_dialog.dart';
+import 'package:risu/components/appbar.dart';
+import 'package:risu/components/outlined_button.dart';
+import 'package:risu/globals.dart';
+import 'package:risu/pages/home/home_page.dart';
+import 'package:risu/pages/home/home_state.dart';
+import 'package:risu/utils/theme.dart';
+
+import 'details_page.dart';
+
+Future<dynamic> getContainerData(
+    BuildContext context, String containerId) async {
+  late http.Response response;
+
+  try {
+    response = await http.post(
+        Uri.parse('http://$serverIp:8080/api/container/details'),
+        headers: <String, String>{
+          'Content-Type': 'application/json; charset=UTF-8'
+        },
+        body: jsonEncode(<String, String>{
+          'containerId': containerId,
+        }));
+  } catch (err) {
```

Comment:
Extend your "try" in the whole function, otherwise it won't catch an error if, for example,the jsonDecode raise an error